subsys/net/buf.c

```diff
@@ -203,6 +203,7 @@ static const struct net_buf_data_cb net_buf_heap_cb = {
 
 const struct net_buf_data_alloc net_buf_heap_alloc = {
 	.cb = &net_buf_heap_cb,
+	.max_alloc_size = K_HEAP_MEM_POOL_SIZE
```

Comment:
minor: add a `,` to the end. It's not needed now, but if this ever gets extended with another struct member, then you just need to add another line instead of also modifying this existing one.

I was also thinking that maybe `0` could have been used to mean "unrestricted" but then again since we do have this define for the system heap I guess we can use that here. Actually, now that I think about it, maybe we anyway want `0` to have this special meaning, since that would give some backwards compatibility to any existing definitions of custom allocators, since any not explicitly initialized struct members are 0 by default.

Reply:
To my mind, using 0 as the special value is great. I was thinking about using a max value but it could confuse someone. The solution with holding an initial value seemed better but was still a lie as in theory this value could be interpreted as maximum contiguous memory block. With 0 it is also easy to use in `net_buf_append_bytes()`.